go/signature/signature_key_templates_test.go

```diff
@@ -19,65 +19,141 @@ import (
+			curveType:   commonpb.EllipticCurveType_NIST_P256,
+			hashType:    commonpb.HashType_SHA256,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
+		{
+			tcName:      "test ECDSA P-384 with SHA512, DER format and RAW output prefix",
+			typeURL:     testutil.ECDSASignerTypeURL,
+			sigTemplate: signature.ECDSAP384KeyWithoutPrefixTemplate(),
+			curveType:   commonpb.EllipticCurveType_NIST_P384,
+			hashType:    commonpb.HashType_SHA512,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
+		{
+			tcName:      "test ECDSA P-521 with SHA512, DER format and RAW output prefix",
+			typeURL:     testutil.ECDSASignerTypeURL,
+			sigTemplate: signature.ECDSAP521KeyWithoutPrefixTemplate(),
+			curveType:   commonpb.EllipticCurveType_NIST_P521,
+			hashType:    commonpb.HashType_SHA512,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
+		{
+			tcName:      "test ED25519 with TINK output prefix",
+			typeURL:     testutil.ED25519SignerTypeURL,
+			sigTemplate: signature.ED25519KeyTemplate(),
+			prefixType:  tinkpb.OutputPrefixType_TINK,
+		},
+		{
+			tcName:      "test ED25519 with RAW output prefix",
+			typeURL:     testutil.ED25519SignerTypeURL,
+			sigTemplate: signature.ED25519KeyWithoutPrefixTemplate(),
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
 	}
-	// ECDSA P-384
-	template = signature.ECDSAP384KeyTemplate()
-	err = checkECDSAKeyTemplate(template,
-		commonpb.HashType_SHA512,
-		commonpb.EllipticCurveType_NIST_P384,
-		ecdsapb.EcdsaSignatureEncoding_DER)
-	if err != nil {
-		t.Errorf("invalid ECDSA P-384 key template: %s", err)
-	}
-	// ECDSA P-521
-	template = signature.ECDSAP521KeyTemplate()
-	err = checkECDSAKeyTemplate(template,
-		commonpb.HashType_SHA512,
-		commonpb.EllipticCurveType_NIST_P521,
-		ecdsapb.EcdsaSignatureEncoding_DER)
-	if err != nil {
-		t.Errorf("invalid ECDSA P-521 key template: %s", err)
+
+	for _, tt := range flagTests {
+		t.Run(tt.tcName, func(t *testing.T) {
+			err := checkECDSAKeyTemplate(tt.sigTemplate,
+				tt.typeURL,
+				tt.hashType,
+				tt.curveType,
+				tt.sigEncoding,
+				tt.prefixType)
+			if err != nil {
+				t.Errorf("failed %s: %s", tt.tcName, err)
+			}
+		})
 	}
 }
 
 func checkECDSAKeyTemplate(template *tinkpb.KeyTemplate,
+	typeURL string,
 	hashType commonpb.HashType,
 	curve commonpb.EllipticCurveType,
-	encoding ecdsapb.EcdsaSignatureEncoding) error {
-	if template.TypeUrl != testutil.ECDSASignerTypeURL {
-		return fmt.Errorf("incorrect typeurl: expect %s, got %s", testutil.ECDSASignerTypeURL, template.TypeUrl)
+	encoding ecdsapb.EcdsaSignatureEncoding,
+	prefixType tinkpb.OutputPrefixType) error {
+	if template.TypeUrl != typeURL {
+		return fmt.Errorf("incorrect typeurl: expect %s, got %s", typeURL, template.TypeUrl)
+	}
+
+	if template.OutputPrefixType.String() != prefixType.String() {
```

Comment:
I think we can compare the prefix type directly, i.e.,

if template.OutputPrefixType != prefixType {
   ...
}

Reply:
Indeed, because  the underlying type of OutputPrefixType is uint32.

Done.